File: app/src/main/java/com/og/emdlc/dagger/AppComponent.java
package com.og.emdlc.dagger;

import com.og.emdlc.activities.MainActivity;

import javax.inject.Singleton;

import dagger.Component;

/**
 * Created by olivier.goutay on 3/31/17.
 */
@Singleton
@Component(modules = {AppModule.class})
public interface AppComponent {

    //Activities
    void inject(MainActivity mainActivity);

    //Tests
    void inject(AbstractTest abstractTest);
}

File: app/src/main/java/com/og/emdlc/dagger/AppComponent.kt
package com.og.emdlc.dagger

import com.og.emdlc.activities.MainActivity
import dagger.Component
import javax.inject.Singleton

/**
 * Created by olivier.goutay on 3/31/17.
 */
@Singleton
@Component(modules = arrayOf(AppModule::class))
interface AppComponent {

    //Activities
    fun inject(mainActivity: MainActivity)

    //Tests
    fun inject(abstractTest: AbstractTest)
}
